File: compose-app/shared/src/commonMain/kotlin/com/sales/app/domain/model/Account.kt
package com.sales.app.domain.model

data class Account(
    val id: Int,
    val name: String,
    val nameFormatted: String,
    val desc: String?,
    val taxationType: Int,
    val taxRate: Int,

    val address: String?,
    val call: String?,
    val whatsapp: String?,
    val footerContent: String?,
    val signature: String?,
    val financialYearStart: String?,
    val country: String? = "India",
    val state: String? = null,
    val taxNumber: String? = null,
    val defaultTaxId: Int? = null,
    val taxApplicationLevel: String = "item",
    val enableDeliveryNotes: Boolean = true,
    val enableGrns: Boolean = true,
    val createdAt: String? = null,
    val updatedAt: String? = null,
    val deletedAt: String? = null
)

File: backend/app/Models/Account.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Account extends Model
{
    use SoftDeletes, \App\Traits\HasLog, \Illuminate\Database\Eloquent\Factories\HasFactory;

    protected $fillable = [
        'name',
        'name_formatted',
        'desc',
        'taxation_type',
        'address',
        'call',
        'whatsapp',
        'footer_content',
        'signature',
        'log_id',
        'financial_year_start',
        'country',
        'state',
        'tax_number',
        'default_tax_id',
        'tax_application_level',
        'visibility',
    ];

    public function defaultTax()
    {
        return $this->belongsTo(Tax::class, 'default_tax_id');
    }

    protected $casts = [
        'signature' => 'boolean',
        'tax_application_level' => 'string', // 'account', 'bill', or 'item'
    ];

    public function users()
    {
        return $this->belongsToMany(User::class, 'user_account_permissions')
            ->withPivot('role')
            ->withTimestamps();
    }

    public function parties()
    {
        return $this->hasMany(Party::class);
    }

    public function items()
    {
        return $this->hasMany(Item::class);
    }
}
